boards/Makefile

```diff
@@ -98,13 +98,28 @@ $(CXXOBJS): %$(OBJEXT): %.cxx
 
 $(BIN): $(OBJS)
 	$(call ARCHIVE, $@, $(OBJS))
+  
+%.dd: %.c
+	$(Q) $(MKDEP) $(DEPPATH) "$(CC)" -- $(CFLAGS) -- $< > $@
+  
+%.ddxx: %.cxx
+	$(Q) $(MKDEP) $(DEPPATH) "$(CC)" -- $(CXXFLAGS) -- $< > $@
+
+makedepfile: $(patsubst %.c,%.dd,$(SRCS))
+	@for f in $^; do cat $$f; done >> Make.dep
+	$(call DELFILE, $^)
+
+makedepxxfile: $(patsubst %.cxx,%.ddxx,$(CXXSRCS))
+	@for f in $^; do cat $$f; done >> Make.dep
+	$(call DELFILE, $^)
 
 .depend: Makefile $(SRCS) $(TOPDIR)$(DELIM).config
+	$(call DELFILE, Make.dep)
```

Comment:
if we still use Make.dep which contain all dependence info as before. The simplest approach is to enhance MKDEP to launch $(CC) in parallel, then we don't touch so many Makefile.

Reply:
But we need to define the new targets as well and as you can see they are slightly different in different cases (according to flags, bin path) and at least the way I found to handle that is to use different dep file extensions for each case (so one target for each).

Reply:
No we don't need, all Makefile can keep as before, in mkdeps:

1. launch gcc in parallel to generate the individual dep file
2. Wait all gcc instance finish, and merge small dep files into the big one

Basically, the change do the same thing in mkdeps.c as you modify in each Makefile.

Reply:
I considered that initially, but in that case we're getting into OS dependant handling of parallelization. Make already handles how to run the jobs in parallel and also deals with the -j flag (also via MAKEFLAGS). We would be redoing a lot of that.
I prefer to do a Make based solution. 
Maybe I can define per-file .dd targets, which would not require one pattern per case (something in the lines of how targets are defined for applications).

Reply:
If you don't want to manage multiple threads manually, we can invoke gcc with all source files once and let gcc do parallel for us.

Reply:
But that is what is doing right now AFAIK. And GCC does not parallelize internally.

Reply:
Yes, you are right.

Reply:
The rule I was mentioning would be something like:
```
$(ASRCS:.S=.dd): %.dd: %.S
	$(Q) $(MKDEP) --obj-suffix $(OBJEXT) $(DEPPATH) "$(CC)" -- $(CFLAGS) -- $< > $@
```
and same for CSRCS, CXXSRCS, HOSTSRCS. I can handle the kernel/non-kernel by having the depend target
inside bin/kbin, passing KDEFINE or not in each case, so that is no issue.

The problem with this, is that if I want to have these rules in Config.mk, I would need to create a template (like SDIR_template) and do `eval` on each Makefile so that these rules are defined after ASRCS and others are already defined. Otherwise, these rules would be evaluated while these variables are empty. 
I think it is not actually worse than the various dep-related rules we have in each Makefile already, but it feels a bit of an ugly solution (specially since I hate having to remember to do this on each new Makefile and will fail silently if you forget).

The only alternative I can think of is to have one dep file extension for each case (eg: .dds, .ddc, .ddx, .ddh) which can then be defined in Config.mk directly. These files are temporary so I don't think it is much of an issue but also feels a bit of a hack.

Reply:
Ok, I think I managed to simplify it quite a bit and looks like a reasonable solution to me. Still needs the same application in all archs.